Add testing helpers in 'internal/assert'.

JivanAmara opened the pull request:
Will be using these in place of testify versions on gpkg provider branch.

gdey commented on internal/assert/assertions.go:
```diff
@@ -0,0 +1,92 @@
+	if !reflect.DeepEqual(expectedValue, actualValue) {
+		msg := fmt.Sprintf("%v != %v (expected)", actualValue, expectedValue)
+		if len(cm) > 0 {
+			msg = msg + " - " + fmt.Sprintf(cm[0].(string), cm[1:]...)
```
You need to check if cm[0] is actually a string. The format of Equal is not idiomatic. One would think Equal works like Println. And not send in a format string.

ARolek replied:
if the first value of `cm` is not a string, this will panic. you need to check the assertion or maybe change the function signature to accept a string then variadic interface{} params.

ARolek commented on internal/assert/assertions.go:
```diff
@@ -0,0 +1,92 @@
+// Log message and fail if actualValue doesn't match expectedValue
+// *Note that DeepEqual will check the value pointed to by two pointers, ie don't use for checking
+//	pointer equality.
+func Equal(t *testing.T, expectedValue interface{}, actualValue interface{}, cm ...interface{}) bool {
```
Can you provide an example of how `cm` should be used? It's not clear from the method signature.

ARolek commented on internal/assert/assertions.go:
```diff
@@ -0,0 +1,92 @@
+	//		{"Error Trace", strings.Join(CallerInfo(), "\n\r\t\t\t")},
+	//		{"Error", failureMessage},
+	//	}
+	_, file, line, _ := runtime.Caller(2)
```
is an error being ignored here? Maybe throw a panic on error if you don't want to return it in this instance?

ARolek commented on internal/assert/assertions.go:
```diff
@@ -0,0 +1,92 @@
+
+// Fail performs t.Errorf(), but modifies the message with the file and line of the caller.
+func Fail(t *testing.T, failureMessage string) bool {
+	//	content := [][2]string{
```
can this code be removed?

ARolek commented on internal/assert/assertions.go:
```diff
@@ -0,0 +1,92 @@
+	//	}
+	_, file, line, _ := runtime.Caller(2)
+	fileParts := strings.Split(file, "/")
+	file = fileParts[len(fileParts)-1]
```
Can `fileparts` ever be 0? if so, this will panic.

JivanAmara replied:
With a check after runtime.Caller(), no, fileparts won't ever be of length 0.

ARolek commented on internal/assert/assertions.go:
```diff
@@ -0,0 +1,92 @@
+	return false
+}
+
+func isNil(object interface{}) bool {
```
maybe name `object` to `val` since an interface{} != struct

JivanAmara commented:
I added some usage examples and extended the custom message args (cm) to take either the args to a Printf() or Print() -like function.  The primary reason for the variable parameter is so people can happily ignore adding a custom message when it's not needed (the majority of cases).

I added a check & panic for the possibility of runtime.Caller() failing.

ARolek commented:
The v0.6.0 branch no longer exists. moving this to be based of the new v0.7.0 branch.
